_dev/src/components/smart-shopping-campaign-creation/smart-shopping-campaign-creation.vue

```diff
@@ -678,41 +666,55 @@ export default {
       );
     },
     setInterfaceForEdition() {
-      let {endDate} = this.foundSsc;
-      const todayYear = new Date().getFullYear();
-
-      if (new Date(endDate).getFullYear() - 10 > todayYear) {
-        endDate = null;
-      }
-      this.campaignName = this.foundSsc.campaignName;
-      this.campaignDurationStartDate = this.foundSsc.startDate;
-      this.campaignDurationEndDate = endDate;
-      this.campaignHasNoProductsFilter = !this.foundSsc.productFilters.length
-        && !this.foundSsc.hasUnhandledFilters;
-      this.campaignDailyBudget = this.foundSsc.dailyBudget;
-      this.campaignIsActive = this.foundSsc.status === CampaignStatus.ELIGIBLE;
-      [this.targetCountry] = this.$options.filters.changeCountriesCodesToNames([
-        this.foundSsc.targetCountry,
-      ]);
-      this.hasUnhandledFilters = this.foundSsc.hasUnhandledFilters;
-      this.debounceName();
-      this.$store.commit(
-        'smartShoppingCampaigns/SET_FILTERS_CHOSEN',
-        this.filtersChosen,
-      );
-      if (this.filtersChosen.length) {
-        let dimensionToEdit = this.$store.state.smartShoppingCampaigns.sscAvailableFilters.find(
-          (dim) => dim.id === this.filtersChosen[0].dimension,
-        );
-        const filtersToFind = this.filtersChosen[0].values;
-        dimensionToEdit = findAndCheckFilter(dimensionToEdit, filtersToFind);
-        this.$store.commit(
-          'smartShoppingCampaigns/SET_DIMENSION_CHOSEN',
-          dimensionToEdit,
-        );
-      }
+      this.$store
+        .dispatch('smartShoppingCampaigns/GET_DIMENSIONS_FILTERS', '')
```

Comment:
Prefer launching this request if there is no filters loaded yet?

Reply:
Also, it could be interesting to send `null` instead of an empty string

Reply:
After trying multiple things, it is the best way to do it because when editing a campaign and refreshing the page we lose the filters if we don't redo the call each time

Reply:
I'm not suggesting to remove it, but to call it only when the filters aren't loaded yet